packages/drivers/odsp-driver/src/odspDocumentStorageManager.ts

```diff
@@ -142,20 +146,51 @@ export class OdspDocumentStorageService implements IDocumentStorageService {
         return "";
     }
 
-    public async createBlob(file: Uint8Array): Promise<api.ICreateBlobResponse> {
-        this.checkSnapshotUrl();
+    public async createBlob(file: IsoBuffer): Promise<api.ICreateBlobResponse> {
+        this.checkAttachmentPOSTUrl();
 
-        return PerformanceEvent.timedExecAsync(
-            this.logger,
-            {
-                eventName: "createBlob",
-                size: file.length,
-            },
-            async () => {
-                // Future implementation goes here
-                // Need to wrap implementation with getWithRetryForTokenRefresh()
-                throw new Error("StandardDocumentStorageManager.createBlob() not implemented");
-            });
+        const response = await getWithRetryForTokenRefresh(async (options) => {
+            const storageToken = await this.getStorageToken(options, "CreateBlob");
+            const { url, headers } = getUrlAndHeadersWithAuth(`${this.attachmentPOSTUrl}/content`, storageToken);
+            headers["Content-Type"] = "application/octet-stream";
+
+            return PerformanceEvent.timedExecAsync(
+                this.logger,
+                {
+                    eventName: "createBlob",
+                    size: file.length,
+                },
+                async () => fetchHelper<api.ICreateBlobResponse>(
+                    url,
+                    {
+                        body: file,
+                        headers,
+                        method: "POST",
+                    },
+                ),
+            );
+        });
+
+        return response.content;
+    }
+
+    public async readBlob(blobid: string) {
+        this.checkAttachmentGETUrl();
+
+        const response = await getWithRetryForTokenRefresh(async (options) => {
+            const storageToken = await this.getStorageToken(options, "ReadDataBlob");
+            const { url, headers } = getUrlAndHeadersWithAuth(`${this.attachmentGETUrl}/${blobid}/content`, storageToken);
+
+            return PerformanceEvent.timedExecAsync(
+                this.logger,
+                {
+                    eventName: "readDataBlob",
+                    headers: Object.keys(headers).length !== 0 ? true : undefined,
+                },
+                async () => fetchHelper<IsoBuffer>(url, { headers }, false),
```

Comment:
Same question as in upload: I believe this is a string, and we need to decode it as it has to be base64.
If it can be in binary format - even better, but it's not a buffer.
In fact, take a look at fetchHelper - it assumes that response is a json payload and it calls JSON.parse() on it.
If it's not json, then we likely need to modify fetchHelper (likely break it into base function that does all the work, and then w user of it - one for json payloads, and one - for binary payloads)